x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenAuthIntegTests.java

```diff
@@ -268,6 +268,23 @@ public void testRefreshingToken() throws IOException {
                 .admin().cluster().prepareHealth().get());
     }
 
+    public void testAuthenticateWithWrongToken() throws IOException {
+        final RestHighLevelClient restClient = new TestRestHighLevelClient();
+        CreateTokenResponse response = restClient.security().createToken(CreateTokenRequest.passwordGrant(
+            SecuritySettingsSource.TEST_USER_NAME, SecuritySettingsSourceField.TEST_PASSWORD.toCharArray()), SECURITY_REQUEST_OPTIONS);
+        assertNotNull(response.getRefreshToken());
+        // First check that the correct access token works by getting cluster health with token
+        assertNoTimeout(client()
+            .filterWithHeader(Collections.singletonMap("Authorization", "Bearer " + response.getAccessToken()))
+            .admin().cluster().prepareHealth().get());
+        // Now attempt to authenticate with an invalid access token string
+        RequestOptions wrongAuthOptions =
+            RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", "Bearer " + randomAlphaOfLengthBetween(0, 128)).build();
+        ElasticsearchStatusException e = expectThrows(ElasticsearchStatusException.class,
+            () -> restClient.security().authenticate(wrongAuthOptions));
+        assertEquals(RestStatus.UNAUTHORIZED, e.status());
```

Comment:
I was expecting you'd invalidate the created token and try another time with it, rather than creating a new random string.

We have a few formats (lengths) for tokens. If you wish to test some/most of them, I would suggest we craft them on purpose, so that we cover the branches in `decodeToken`. I think a long random string is insufficient to cover for "invalid" tokens, generally.

If you only test the REST response code, which I believe you are aiming for, a random string is enough (no need to create another token).